--- origsrc/rpcbind-0.2.4/src/rpcb_svc_com.c	2017-06-27 02:19:33.344600800 -0500
+++ src/rpcbind-0.2.4/src/rpcb_svc_com.c	2017-06-27 02:21:38.540761800 -0500
@@ -43,7 +43,9 @@
 #include <sys/stat.h>
 #include <sys/param.h>
 #include <sys/poll.h>
+#ifndef __CYGWIN__
 #include <bits/poll.h>
+#endif
 #include <sys/socket.h>
 #include <rpc/rpc.h>
 #include <rpc/rpcb_prot.h>
@@ -67,8 +69,12 @@
 #define RPC_BUF_MAX	65536	/* can be raised if required */
 
 
+#ifndef POLLRDNORM
 # define POLLRDNORM	0x040		/* Normal data may be read.  */
+#endif
+#ifndef POLLRDBAND
 # define POLLRDBAND	0x080		/* Priority data may be read.  */
+#endif
 
 
 
--- origsrc/rpcbind-0.2.4/src/rpcbind.c	2016-11-28 14:47:28.000000000 -0600
+++ src/rpcbind-0.2.4/src/rpcbind.c	2017-06-27 02:21:38.547761800 -0500
@@ -94,7 +94,7 @@ rpcblist_ptr list_rbl;	/* A list of vers
 #ifdef RPCBIND_USER
 char *rpcbinduser = RPCBIND_USER;
 #else
-char *rpcbinduser = NULL;
+char *rpcbinduser = "";
 #endif
 
 #ifdef NSS_MODULES
@@ -104,7 +104,11 @@ char *nss_modules = "files";
 #endif
 
 /* who to suid to if -s is given */
+#ifdef __CYGWIN__
+#define RUN_AS  "SYSTEM"
+#else
 #define RUN_AS  "daemon"
+#endif
 
 #define RPCBINDDLOCK "/var/run/rpcbind.lock"
 
@@ -171,10 +175,12 @@ main(int argc, char *argv[])
 		setrlimit(RLIMIT_NOFILE, &rl);
 	}
 
+#ifndef __CYGWIN__
 	if (geteuid()) { /* This command allowed only to root */
 		fprintf(stderr, "Sorry. You are not superuser\n");
 		exit(1);
 	}
+#endif
 
 	/*
 	 * Make sure we use the local service file 
@@ -249,7 +255,7 @@ main(int argc, char *argv[])
         	err(1, "fork failed");
 	}
 
-	if (runasdaemon || rpcbinduser) {
+	if (runasdaemon || *rpcbinduser) {
 		struct passwd *p;
 		char *id = runasdaemon ? RUN_AS : rpcbinduser;
 
--- origsrc/rpcbind-0.2.4/src/security.c	2016-11-28 14:47:28.000000000 -0600
+++ src/rpcbind-0.2.4/src/security.c	2017-06-27 02:21:38.554262800 -0500
@@ -72,7 +71,7 @@ int deny_severity = LIBWRAP_DENY_FACILIT
 #ifndef PORTMAP_LOG_SEVERITY
 # define PORTMAP_LOG_SEVERITY LOG_INFO
 #endif
-#ifdef __linux__
+#ifndef IPV6PORT_RESERVED
  #define IPV6PORT_RESERVED IPPORT_RESERVED
 #endif
 
